### PR TITLE
docs: add more build examples 

### PR DESCRIPTION
**Pull request by levlaz**
* add cross compile examples to build section 
* Add Dockerfile examples to build section
* Add Dockerfile with different build context example

### CODE REVIEW

**Comment by jpadams** [comment]
Cross-compilation
```
cross-compile ➤ cd go
go ➤ dagger functions
21:10:32 WRN masked parent span not found; possible data loss? traceID=00000000000000000000000000000000 spanID=636baf8e9638f23d parentID=b80db86a4a8c4497
21:10:33 WRN masked parent span not found; possible data loss? traceID=00000000000000000000000000000000 spanID=636baf8e9638f23d parentID=b80db86a4a8c4497
21:10:33 WRN masked parent span not found; possible data loss? traceID=00000000000000000000000000000000 spanID=636baf8e9638f23d parentID=b80db86a4a8c4497

Name    Description
build   Build and publish multi-platform image

go ➤ dagger version
dagger v0.11.6 (registry.dagger.io/engine) darwin/arm64
```

File: docs/current_docs/manuals/developer/cookbook/snippets/builds/multi-arch-cc/go/main.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			WithExec([]string{"go", "build", "-o", "/output/hello"})
+
+		// select output directory
+		outputDir := ctr.Directory("/output")
+
+		// wrap the output directory in the new empty container marked
+		// with the same platform
+		binaryCtr := dag.Container(ContainerOpts{Platform: platform}).
+			WithRootfs(outputDir)
+
```
Comment (jpadams) [inline docs/current_docs/manuals/developer/cookbook/snippets/builds/multi-arch-cc/go/main.go:61]:
```suggestion
						WithEntrypoint([]string{"/hello"})
```

File: docs/current_docs/manuals/developer/cookbook/snippets/builds/multi-arch-cc/go/main.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+			WithWorkdir("/src").
+			WithExec([]string{"go", "build", "-o", "/output/hello"})
+
+		// select output directory
+		outputDir := ctr.Directory("/output")
+
+		// wrap the output directory in the new empty container marked
+		// with the same platform
+		binaryCtr := dag.Container(ContainerOpts{Platform: platform}).
+			WithRootfs(outputDir)
```
Comment (jpadams) [inline docs/current_docs/manuals/developer/cookbook/snippets/builds/multi-arch-cc/go/main.go:60]:
```suggestion
			WithRootfs(outputDir).
```

**Review by jpadams — APPROVED** [review]
SUMMARY:
All of the examples work, but I'd definitely add the `WithEntrypoint()` to the multi arch ones. The rest of the advice is optional, since things all technically work. I understand the challenge of getting to the heart of the matter with a short example versus showing a complete example.

The advice on the Dockerfile example is to ensure we have a running _and functioning_ container, but it does build and run, just not function out of the box. I suppose someone could go follow the original example and use the built image with it, so again, it's technically correct. I suggested a different name for the container image as well.

Could use more descriptive or generic name for the container image rather than `hello-dagger` for the Dockerfile with alt context example.

File: docs/current_docs/manuals/developer/cookbook/snippets/builds/dockerfile/go/main.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	ctx context.Context,
+	// location of directory containing Dockerfile
+	src *Directory,
+) (string, error) {
+	ref, err := dag.Container().
+		WithDirectory("/src", src).
+		WithWorkdir("/src").
+		Directory("/src").
+		DockerBuild().
+		Publish(ctx, "ttl.sh/hello-dagger")
```
[the remark this answers is not in the document]
Comment (levlaz) [reply docs/current_docs/manuals/developer/cookbook/snippets/builds/dockerfile/go/main.go:20]: This is excellent feedback, I spun this out in to a follow on issue

**Pull request merged**